### PR TITLE
docs: improve .golangci.reference.yml defaults

### PR DESCRIPTION
**Pull request by mateusoliveira43**
Add some defaults to `.golangci.reference.yml` file.

Add color to all flag descriptions.

### CODE REVIEW

File: pkg/commands/root.go [hunk 1]
```diff
@@ -119,8 +119,9 @@ func formatMemory(memBytes uint64) string {
 
 func getDefaultConcurrency() int {
 	if os.Getenv(envHelpRun) == "1" {
-		// Make default concurrency stable: don't depend on machine CPU number, for docs/src/docs/usage/configuration.mdx help generating builds.
-		return 0
+		// Make stable concurrency for README help generating builds.
```
Comment (mateusoliveira43) [inline pkg/commands/root.go]: I think this change is misleading. Earlier, the script would populate README file, but now it only populates files in `docs/` folder

And changing returning to non zero (non nil value) will print default (8). User may think the default is 8 when seeing the the website docs

File: pkg/commands/root.go [hunk 2]
```diff
@@ -170,5 +171,4 @@ func initRootFlagSet(fs *pflag.FlagSet, cfg *config.Config, needVersionOption bo
 	}
 
 	fs.StringVar(&cfg.Output.Color, "color", "auto", wh("Use color when printing; can be 'always', 'auto', or 'never'"))
-	fs.BoolP("help", "h", false, wh("Help for this command"))
```
Comment (mateusoliveira43) [inline pkg/commands/root.go]: Without this, help commands does not get colored
Screenshot from 2023-12-19 17-06-04

Comment (ldez) [reply pkg/commands/root.go]: Yes It's expected.

**Review by ldez — APPROVED** [review]
2 topics = 2 PRs
Please don't create PR with multiple topics.
I will merge the PR in this state (exceptionally) but please avoid that.

**Pull request merged**